File: kora-java-crud/src/main/java/ru/tinkoff/kora/example/crud/model/mapper/PetMapper.java
package ru.tinkoff.kora.example.crud.model.mapper;

import org.mapstruct.Mapper;
import ru.tinkoff.kora.example.crud.model.dao.PetCategory;
import ru.tinkoff.kora.example.crud.model.dao.PetWithCategory;
import ru.tinkoff.kora.example.crud.openapi.http.server.model.CategoryTO;
import ru.tinkoff.kora.example.crud.openapi.http.server.model.PetTO;

@Mapper
public interface PetMapper {

    PetTO asDTO(PetWithCategory pet);

    CategoryTO asDTO(PetCategory category);
}

File: kora-kotlin-crud/src/main/kotlin/ru/tinkoff/kora/kotlin/example/crud/model/PetMapper.kt
package ru.tinkoff.kora.kotlin.example.crud.model

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import ru.tinkoff.kora.example.crud.openapi.http.server.model.CategoryTO
import ru.tinkoff.kora.example.crud.openapi.http.server.model.PetTO

@Mapper
interface PetMapper {

    fun petWithCategoryToPetTO(pet: PetWithCategory): PetTO

    fun petCategoryToCategoryTO(category: PetCategory): CategoryTO
}
